### DISCUSSION TITLE
PS5 controller stick is way too sensitive for straight angles and directions for n64 is there anyway to change?

### DISCUSSION THREAD

**Discussion by as4pu** [darthcloud/BlueRetro#926] [Q&A]
By sensitive I don't mean by holding a little bit down means it goes all the way no, i mean mostly it's very hard to hold straight up as it's just too responsive and it's very hard to consistently get a straight up angle as it mostly does straight up but with a little bit to the right or a little bit to the left or it wiggles too much unless i leave my thumb PERFECTLY still in that direction. (this goes for all angles and directions such as up, down, left, right, up-right, etc). Yes it's possible to get a straight up angle but it just has to be way too precise unlike on Project64 on the PC it's MUCH easier to hold straight on an angle/direction as it doesn't feel that precise and it's way more lenient. 

I'm asking this because I normally speed run SM64 on emulator with this ps5 controller for a while now and I've never had this issue on Project64 as using a ps5 controller on Project64 feels super smooth and not hard to get angles/directions at all. I've tried messing around with every setting in the blueretro.io advanced config, Max, Threshold, deadzone, scaling etc and all it does is change the range of the stick (by changing the Max to lower % for example, it makes all the way up on the stick move mario slower but it's still just as non-lenient with angles/directions). I'm starting to think it might not be possible but maybe there's some secret combo of settings that will just make it as smooth as project64.. I use N-Rage Input Plugin V2 on project64 for my controller with an analog stick Range percent of 70% with Real N64 Range unchecked and a deadzone of 10%. 

I love how this blueretro adapter doesn't have any noticeable lag or at MOST a frame of lag which is fine and I really dislike the feel of an OEM n64 controller and controllers with notches in general, but this basically makes it impossible as you have to do straight angles/directions a lot to do things correctly. I hope I made sense and thanks for reading hopefully someone helps me out :)

**Comment by darthcloud** [comment]
For now everything is converted as if the range of both the source and destination were linear.

Which is not true in reality but that still good enough for most people.

Also the diagonal maximal range isn't accounted for since n64 joystick gate is a square-ish hexagon and most modern ctrl are either a circle or an equilateral hexagon.

I certainly want to address that in the future.
But with the little time I got, I mostly focussing on smaller task.
